src/components/Icon/tests/Icon.test.tsx

```diff
@@ -1,12 +1,280 @@
 import * as React from 'react';
-import { shallow } from 'enzyme';
-import Icon from '../Icon';
+import { mount } from 'enzyme';
+import Icon from '..';
+
+const theme = {
+  Icon: 'Icon',
+  hasBackdrop: 'hasBackdrop',
+  colorWhite: 'colorWhite',
+  Svg: 'Svg',
+  Placeholder: 'Placeholder',
+};
 
 describe('<Icon />', () => {
-  describe('accessibilityLabel', () => {
-    it('uses the label as the aria-label for the icon', () => {
-      const element = shallow(<Icon source="placeholder" accessibilityLabel="This is an icon" />);
-      expect(element.prop('accessibilityLabel')).toBe('This is an icon');
+
+  describe('when default props are provided', () => {
+    it('basic icon should have rendered one span element', () => {
+      const iconWrapper = mount(
+                                <Icon source="notes" theme={theme} />,
+                          );
+      expect(iconWrapper.find('span')).toHaveLength(1);
+    });
+    it('basic icon should have rendered with one icon clss on span', () => {
+      const iconWrapper = mount(
+                                <Icon source="notes" theme={theme} />,
+                          );
+      expect(iconWrapper.find('span').hasClass('Icon'));
+    });
+  });
+
+  describe('source property', () => {
+    describe('when set as svg source', () => {
+      it('basic icon should have rendered one span element', () => {
+        const iconWrapper = mount(
+                                  <Icon source="notes" theme={theme} />,
+                            );
+        expect(iconWrapper.find('span')).toHaveLength(1);
+      });
+      it('basic icon should have rendered with one icon clss on span', () => {
+        const iconWrapper = mount(
+                                  <Icon source="notes" theme={theme} />,
+                            );
+        expect(iconWrapper.find('span').hasClass('Icon'));
+      });
+      it('should verify given source', () => {
+        const iconWrapper = mount(
+                                  <Icon source="notes" theme={theme} />,
+                            );
+        expect(iconWrapper.prop('source')).toBe('notes');
+      });
+    });
+
+      describe('when set as placeholder', () => {
+        it('basic icon should have rendered one span element', () => {
+          const iconWrapper = mount(
```

Comment:
you have the same constant iconWrapper declared 5 times. Why not just declare it once after line 52?